OpenRA.Mods.Common/Traits/Conditions/GrantConditionAfterDelay.cs

```diff
@@ -0,0 +1,103 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Gives a condition to the actor after a delay.")]
+	public class GrantConditionAfterDelayInfo : PausableConditionalTraitInfo
```

Comment:
I suspect that the best fix will be to invert the logic, and implement this as a `GrantTimedCondition` trait, or maybe via a `Duration` field on `GrantCondition` + the existing `TimedConditionBar` trait.

Reply:
I'm usnsure about this, i still think as we talked before getting a WithSelectionBarInfo with a bool to reverse it sounds like a good idea. Would be nice for unifying the traits around that. (We can do same with IPipInfo).

For complateness of this PR i can implement that only for this trait here, and we can adopt other traits later.